ledger/ledger-api-common/src/main/scala/com/digitalasset/platform/server/api/validation/ErrorFactories.scala

```diff
@@ -4,54 +4,204 @@
 package com.daml.platform.server.api.validation
 
 import com.daml.ledger.api.domain.LedgerId
+import com.daml.ledger.grpc.GrpcStatuses
 import com.daml.platform.server.api.ApiException
-import io.grpc.{Status, StatusRuntimeException}
-
+import com.google.protobuf.{Any => AnyProto}
+import com.google.rpc.{ErrorInfo, Status}
+import io.grpc.Status.Code
+import io.grpc.StatusRuntimeException
+import io.grpc.protobuf.StatusProto
 import scalaz.syntax.tag._
 
 trait ErrorFactories {
 
-  def ledgerIdMismatch(expected: LedgerId, received: LedgerId): StatusRuntimeException =
+  import ErrorFactories._
+
+  lazy val DuplicateCommandException: StatusRuntimeException =
     grpcError(
-      Status.NOT_FOUND.withDescription(
-        s"Ledger ID '${received.unwrap}' not found. Actual Ledger ID is '${expected.unwrap}'."
-      )
+      Status
+        .newBuilder()
+        .setCode(Code.ALREADY_EXISTS.value())
+        .setMessage("Duplicate command")
+        .addDetails(definiteAnswers(true))
```

Comment:
```suggestion
        .addDetails(definiteAnswers(false))
```
As explained in https://github.com/digital-asset/daml/pull/10801#discussion_r705956197, we don't persist anything for duplicate commands and therefore that's not a definite answer.